### PR TITLE
Added icinga2 module

### PR DESCRIPTION
**Pull request by ManiacTwister**

### CODE REVIEW

File: input/icinga2.go [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+}
+
+type hgmapping struct {
+	DefaultChannel    string              `mapstructure:"default"`
+	HostGroupMappings map[string][]string `mapstructure:"hostgroups"`
+	ExplicitMappings  map[string][]string `mapstructure:"explicit"`
+}
+
+func (m *Icinga2Module) Init(c *viper.Viper, channel *chan IRCMessage) {
+	err := c.UnmarshalKey("default", &m.channelMapping.DefaultChannel)
```
Comment (fleaz) [inline input/icinga2.go]: You don't need three unmarshalKey() calls and can use just one like for the GitLab module https://github.com/fleaz/CptHook/blob/master/input/gitlab.go#L37

File: input/icinga2.go [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+				m.sendMessage(buf.String(), notification)
+			} else { // State changed
+				err = hostStateChangeTemplate.Execute(&buf, &notification)
+				m.sendMessage(buf.String(), notification)
+				buf.Reset()
+				err = hostOutputTemplate.Execute(&buf, &notification)
+				m.sendMessage(buf.String(), notification)
+			}
+		default:
+			log.Printf("[icinga2] Unknown event:")
```
Comment (fleaz) [inline input/icinga2.go]: The event itself is missing here ;)

File: input/icinga2.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
+package input
+
+import (
+	"time"
+	"bytes"
+	"strings"
+	"net/http"
+	"text/template"
+	"encoding/json"
+	"github.com/dustin/go-humanize"
```
Comment (fleaz) [inline input/icinga2.go]: This is a new dependency. Could you please run `dep` to also update the Lockfile

File: input/icinga2.go (hunk 1 above)
Comment (ManiacTwister) [reply input/icinga2.go]: The problem with unmarshal() is that keys with dots aren't working but we need them for hostnames. 
https://github.com/fleaz/CptHook/pull/31#issuecomment-436089074

Comment (fleaz) [reply input/icinga2.go]: Ah yes, there was an issue. I remember that. Still unfixed upstream :(

**Pull request merged**